benchmark/src/DriverBench/ParallelBench.php

```diff
@@ -0,0 +1,199 @@
+            unlink($file);
+        }
+
+        self::$files = [];
+    }
+
+    /**
+     * Parallel: LDJSON multi-file import
+     * Using single thread
+     */
+    #[BeforeMethods('beforeMultiFileImport')]
+    #[Revs(1)]
+    public function benchMultiFileImport(): void
+    {
+        $collection = Utils::getCollection();
+        foreach (self::getFileNames() as $file) {
+            self::importFile($file, $collection);
+        }
+    }
+
+    /**
+     * Parallel: LDJSON multi-file import
+     * Using multiple forked threads
+     *
+     * @param array{processes:int, files:string[], batchSize:int} $params
+     */
+    #[BeforeMethods('beforeMultiFileImport')]
+    #[ParamProviders(['provideProcessesParameter'])]
+    #[Revs(1)]
+    public function benchMultiFileImportFork(array $params): void
+    {
+        $pids = [];
+        foreach (self::getFileNames() as $file) {
+            // Wait for a child process to finish if we have reached the maximum number of processes
+            if (count($pids) >= $params['processes']) {
+                $pid = pcntl_waitpid(-1, $status);
+                unset($pids[$pid]);
+            }
+
+            $pid = pcntl_fork();
+            if ($pid === 0) {
+                // If we reset, we can garantee that we get a new manager in the child process
+                // If we don't reset, we will get the same manager client_zval in the child process
+                // and share the libmongoc client.
+                Utils::reset();
+                self::importFile($file, Utils::getCollection());
+
+                // Exit the child process
+                exit(0);
+            }
+
+            if ($pid === -1) {
+                throw new RuntimeException('Failed to fork');
+            }
+
+            // Keep the forked process id to wait for it later
+            $pids[$pid] = true;
+        }
+
+        // Wait for all child processes to finish
+        while ($pids !== []) {
+            $pid = pcntl_waitpid(-1, $status);
+            unset($pids[$pid]);
+        }
+    }
+
+    /**
+     * Parallel: LDJSON multi-file import
+     * Using amphp/parallel-functions with worker pool
+     *
+     * @param array{processes:int, files:string[], batchSize:int} $params
+     */
+    #[BeforeMethods('beforeMultiFileImport')]
+    #[ParamProviders(['provideProcessesParameter'])]
+    #[Revs(1)]
+    public function benchMultiFileImportAmp(array $params): void
+    {
+        wait(parallelMap(
+            self::getFileNames(),
+            // Uses array callable instead of closure to skip complex serialization
+            [self::class, 'importFile'],
+            // The pool size is the number of processes
+            new DefaultPool($params['processes']),
+        ));
+    }
+
+    public static function provideProcessesParameter(): Generator
+    {
+        // Max number of forked processes
+        for ($i = 1; $i <= 30; $i = (int) ceil($i * 1.25)) {
```

Comment:
What do you think of trying exponential increase?
```suggestion
        for ($i = 2; $i <= 32; $i *= 2) {
```

That would run 5 benchmarks (with 2, 4, 8, 16, and 32 processes) instead of the current 12 benchmarks and thus save a bunch of time in addition to the savings we'd get from only running a single iteration.

Reply:
I preferred to be verbose so I can set a number of processes that matches with the number of round trip that are necessary to handle 100 files.